code/statistics/model/studentmarks.py

```diff
@@ -0,0 +1,62 @@
+class StudentMarks(object):
+    @staticmethod
+    def is_correct(list_of_marks):
+        for mark in list_of_marks:
+            if not isinstance(mark, int):
+                raise TypeError
+            if mark not in [1, 2, 3, 4, 5]:
+                raise ValueError
+        return True
+
+    def __init__(self, array_of_marks: list):
+        if StudentMarks.is_correct(array_of_marks):
+            self.marks = array_of_marks
+
+    def amount_of_marks(self):
+        return len(self.marks)
+
+    def average_of_marks(self):
+        summary = 0
+        count = self.amount_of_marks()
+        for i in self.marks:
+            summary += i
+        return summary / count
+
+
+class Register:
+    def __init__(self, students: list):
+        self.students = students
+
+    def add_new_student(self, student):
+        if StudentMarks.is_correct(student):
+            self.students.append(student)
+
+    def surnames_of_losers(self):
+        surnames = []
+        for student in self.students:
+            if student.average_of_marks() < 3:
+                surnames.append(student)
+        return surnames
+
+    def count_of_losers(self):
+        return len(self.surnames_of_losers())
+
+    def surnames_of_excellent_students(self):
+        surnames = []
+        for student in self.students:
+            if student.average_of_marks() >= 4.5:
```

Comment:
Same values 4.5 in methods "surnames_of_excellent_students" and "students_who_successfully_pass"

Reply:
Oh, thanks, you are absolutely right